api/storage_service.cc

```diff
@@ -1428,8 +1428,10 @@ void set_snapshot(http_context& ctx, routes& r, sharded<db::snapshot_ctl>& snap_
                     return cm.perform_sstable_scrub(cf.as_table_state(), opts);
                 }, std::make_optional(sstables::compaction_counters{}), reduce_compaction_counters);
             }, std::make_optional(sstables::compaction_counters{}), reduce_compaction_counters);
-        }).then([] (auto f) {
-            if (f.value().validation_errors == 0) {
+        }).then_wrapped([] (auto f) {
+            if (f.failed()) {
+                return make_ready_future<json::json_return_type>(static_cast<int>(scrub_status::aborted));
```

Comment:
Let's keep this status code only for `compaction_aborted_exception` so if scrub failed for any other reason (e.g. bad_alloc) the failure would still be represented as an error to execute the api call - since the user doesn't know in this case if scrub even ran. Maybe it failed early and didn't even get to scrub any sstable - hence the error is inconclusive.

you can use something like:
```c++
    if (f.failed()) {
        auto ex = f.get_exception()
        if (try_catch<sstables::compaction_aborted_exception>(ex)) {
                return make_ready_future<json::json_return_type>(static_cast<int>(scrub_status::aborted));
        } else {
                return make_exception_future<json::json_return_type>(std::move(ex));
        }
    }
```

That said, note that the code in this patch drops the exception without officially handling it, i.e. it never calls f.get_exception().
We have a sanity check in seastar to report unhandled exception in future so when the future is destroyed while hold an exception seastar will warn about it, JSYK :)